webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ReportsJdbcRepository.scala

```diff
@@ -119,12 +119,12 @@ class ReportsJdbcRepository(doobie: Doobie) extends ReportsRepository with Logga
 
   override def findReportsByNodeOnInterval(
       nodeId: NodeId
-    , start : DateTime
-    , end   : DateTime
+    , start : Option[DateTime]
+    , end   : Option[DateTime]
   ) : Vector[Reports] = {
     val q = Query[(NodeId, DateTime, DateTime), Reports](baseQuery +
```

Comment:
Only idea i got is pattern matching here 

val q = (start,end) match {
case (Some(start), Some(end)) => Query NodeId,DateTime, DateTime
case (None,Some(end)) => Query NodeId,DateTime
case (Some(start),None))=> Query NodeId,DateTime
case (None, None) => Query NodeId
}

I leave you the pleasure to build the queries!!